src/app/shared/components/Breadcrumbs/BreadcrumbsMenu/BreadcrumbsMenu.tsx

```diff
@@ -0,0 +1,283 @@
+import { Menu, Transition } from '@headlessui/react';
+import {
+  CaretDown,
+  FolderSimplePlus,
+  Trash,
+  PencilSimple,
+  Link,
+  DownloadSimple,
+  Users,
+  Info,
+} from '@phosphor-icons/react';
+import { ReactComponent as MoveActionIcon } from 'assets/icons/move.svg';
+import { useTranslationContext } from 'app/i18n/provider/TranslationProvider';
+import moveItemsToTrash from '../../../../../use_cases/trash/move-items-to-trash';
+import { DriveItemData, DriveItemDetails } from 'app/drive/types';
+import { sharedThunks } from 'app/store/slices/sharedLinks';
+import { downloadItemsThunk } from 'app/store/slices/storage/storage.thunks/downloadItemsThunk';
+import { useAppDispatch, useAppSelector } from 'app/store/hooks';
+import { uiActions } from '../../../../store/slices/ui';
+import useDriveItemStoreProps from 'app/drive/components/DriveExplorer/DriveExplorerItem/hooks/useDriveStoreProps';
+import { getAppConfig } from 'app/core/services/config.service';
+import { BreadcrumbItemData } from '../types';
+import storageThunks from 'app/store/slices/storage/storage.thunks';
+import { storageActions } from '../../../../store/slices/storage';
+
+interface BreadcrumbsMenu {
+  item: BreadcrumbItemData;
+  items: BreadcrumbItemData[];
+  onItemClicked: (item: BreadcrumbItemData) => void;
+}
+
+const BreadcrumbsMenu = (props: BreadcrumbsMenu) => {
+  const { onItemClicked } = props;
+  const { translate } = useTranslationContext();
+  const dispatch = useAppDispatch();
+  const allItems = useAppSelector((state) => state.storage.levels);
+  const namePath = useAppSelector((state) => state.storage.namePath);
+  const currentBreadcrumb = namePath[namePath.length - 1];
+  const { breadcrumbDirtyName } = useDriveItemStoreProps();
+  const currentDevice = useAppSelector((state) => state.backups.currentDevice);
+  const path = getAppConfig().views.find((view) => view.path === location.pathname);
+  const pathId = path?.id;
+  const isSharedView = pathId === 'shared';
```

Comment:
I would do the same as you have done with the other containers in this Menu, I would pass the menu to the BreadcbrumbsItem by props instead of importing it directly, so in each container you can customize the BreadcrumbMenu and let the breadcrumb container of each view dictate the behaviour of the menu and the breadcrumbs.
That way the component will be simpler and will have less logic that belongs to each view where it is implemented.

Reply:
Done